src/pandantic/validators/pandas.py

```diff
@@ -0,0 +1,132 @@
+import logging
+import math
+import os
+from typing import Any, Optional
+
+import pandas as pd
+from multiprocess import (  # type:ignore # pylint: disable=no-name-in-module
+    Process,
+    Queue,
+)
+
+from pandantic.types import SchemaTypes
+from pandantic.validators.baseclass import BaseValidator
+
+
+class PandasValidator(BaseValidator):
+    def __init__(self, schema: SchemaTypes):
+        self.schema = schema
+
+    def validate(
+        self,
+        dataframe: pd.DataFrame,
+        errors: str = "raise",
+        context: Optional[
+            dict[str, Any]
+        ] = None,  # pylint: disable=consider-alternative-union-syntax,useless-suppression
+        n_jobs: int = 1,
+        verbose: bool = True,
+    ) -> pd.DataFrame:
+        """Validate a DataFrame using the schema defined in the Pydantic model.
+
+        Args:
+            dataframe (pd.DataFrame): The DataFrame to validate.
+            errors (str, optional): How to handle validation errors. Defaults to "raise".
+            context (Optional[dict[str, Any], None], optional): The context to use for validation.
+            n_jobs (int, optional): The number of processes to use for validation. Defaults to 1.
+            verbose (bool, optional): Whether to log validation errors. Defaults to True.
+
+        Returns:
+            pd.DataFrame: The DataFrame with valid rows in case of errors="filter".
+        """
+        errors_index = []
+        logging.debug("Amount of available cores: %s", os.cpu_count())
+
+        dataframe = dataframe.copy()
+        dataframe["_index"] = dataframe.index
+
+        if n_jobs != 1:
+            if n_jobs < 0:
+                n_jobs = os.cpu_count()  # type: ignore
+
+            chunks = []
+            chunk_size = math.floor(len(dataframe) / n_jobs)
+            num_chunks = len(dataframe) // chunk_size + 1
+
+            q = Queue()
```

Comment:
We should have the Queue as an optional argument. One might want to abstract the multiprocessing aspect, or run an async Queue or something like that. The default should remain as is.

Reply:
Moved it to the arguments as an optional Queue object. Will only be defaulted to Queue if nothing is specified in the scope of the multiprocessing run in case of the amount of jobs is not 1. 

I was wondering, should I change the typing to Any as we want to allow multiple Queue classes from multiple packages?